doc/getting-started.rst

```diff
@@ -222,6 +222,36 @@ can be run directly after setting the execute permission (+x):
 
 See more in the documentation for :ref:`cabal run`.
 
+.. warning::
+
+    Single-file scripts cannot also be part of a package, as an executable or
+    listed as a module.
+
+    .. code-block:: console
+
+        $ cat script-exclusivity.cabal
+        cabal-version: 3.0
+        name: script-exclusitivity
+        version: 1
+
+        executable my-script-exe
+            build-depends:
+                base,
+                haskell-say
+            main-is: myscript.hs
+
+        $ ./myscript.hs
+        Error: [Cabal-7070]
+        The run command can only run an executable as a whole, not files or modules
+        within them, but the target 'myscript.hs' refers to the file myscript.hs in the
+        executable my-script-exe.
+
+        $ cabal run myscript.hs
+        Error: [Cabal-7070]
+        The run command can only run an executable as a whole, not files or modules
+        within them, but the target 'myscript.hs' refers to the file myscript.hs in the
+        executable my-script-exe.
+
```

Comment:
I think this code block is pointless and only takes space. The text of the warning above is clear enough. Hence, my suggestion:
```suggestion
```

Reply:
As an alternative, perhaps this explanation should also be on `errors.haskell.org`?

Reply:
Oh, I love this idea!

Reply:
> I think this code block is pointless and only takes space.

@ulysses4ever, I thought this behaviour was suprising enough to warrant calling it out explicitly. It is as if the run command does not recognize this file is also self-contained script, #10324?

Reply:
Calling it out isn't the question; the example isn't really suitable for the user guide, though. If the warning isn't good enough without the code block, it's the warning that should be made more explicit.

Reply:
I know we'd like to see our errors on the [error index](https://errors.haskell.org/) but they're not there yet, are they? If they were I could just link to [Cabal-7070](https://errors.haskell.org/messages/Cabal-7070) instead of this long-winded example that shows two ways to get the same error.

Reply:
> it's the warning that should be made more explicit

I had thought of this addition to the user guide as a stop gap measure until we refine the behaviour.

Reply:
Looks like they aren't, as yet. This might be a good starting place, though.

Reply:
I've added https://github.com/haskellfoundation/error-message-index/pull/536.

Reply:
I'm now linking to the error index and moved the example over there.